data/mods/Russ_Extra_Stuff/items/armor.json

```diff
@@ -0,0 +1,138 @@
+        "cutting" : 0,
+        "warmth" : 0,
+        "phase" : "solid",
+        "environmental_protection" : 0,
+        "encumbrance" : 0,
+        "bashing" : 0,
+	   "piercing" : 0,
+        "flags" : ["WAIST","WATERPROOF","STURDY","OUTER","OVERSIZE"],
+        "coverage" : 0,
+        "material_thickness" : 2,
+"max_charges": 100,
+    "initial_charges": 100,
+    "charges_per_use": 1,
+    "turns_per_charge": 0,
+    "ammo": "battery",
+    "revert_to": "null",
+    "use_action": {
+        "type": "transform",
+        "msg": "You power up the shield generator.",
+        "target": "shield_generator_on",
+        "active": true,
+        "need_charges": 1,
+        "need_charges_msg": "The shield generator's batteries are dead."
+			}
+
+   },{
+
+        "type" : "TOOL_ARMOR",
+        "id" : "shield_generator_on",
+        "name" : "shield generator (on)",
+        "weight" : 840,
+        "color" : "light_blue",
+        "covers" :["TORSO","HEAD","LEGS","ARMS","HANDS","EYES","MOUTH","FEET"],
+        "to_hit" : 0,
+        "storage" : 0,
+        "symbol" : "o",
+        "description" : "Your typical shield generator, reminiscent of the 90's game Out Of This World.  Runs on batteries.  It has a radiation compensation unit built in, but running it generates a lot of heat. (activate again to use)",
+        "price" : 100000,
+        "material" : ["shieldgen", "null"],
+        "volume" : 2,
+        "cutting" : 0,
+        "warmth" : 15,
+        "phase" : "solid",
+        "environmental_protection" : 5,
+        "encumbrance" : 0,
+        "bashing" : 0,
+	   "piercing" : 0,
+        "flags" : ["WATERPROOF","STURDY","OUTER","OVERSIZE"],
+        "coverage" : 100,
+        "material_thickness" : 5,
+"max_charges": 100,
+    "initial_charges": 100,
+    "charges_per_use": 0,
+    "turns_per_charge": 30,
+    "ammo": "battery",
+    "revert_to": "shield_generator",
+    "use_action": {
+        "type": "transform",
+        "msg": "You turn the sheild generator's radiation filter on.",
+        "target": "shield_generator_on_heating",
+	   "active": true,
+        "need_charges": 1
+			}
+    },{
+
+        "type" : "TOOL_ARMOR",
+        "id" : "shield_generator_on_heating",
+        "name" : "shield generator (on, anti-rad)",
+        "weight" : 840,
+        "color" : "light_blue",
+        "covers" :["TORSO","HEAD","LEGS","ARMS","HANDS","EYES","MOUTH","FEET"],
+        "to_hit" : 0,
+        "storage" : 0,
+        "symbol" : "o",
+        "description" : "Your typical shield generator, reminiscent of a 90's nintendo game you used to play called Out Of This World.  Runs on batteries.  It has a radiation compensation unit which is currently running, but it causes the EM field to generate a lot of heat around your body.",
+        "price" : 100000,
+        "material" : ["shieldgen", "null"],
+        "volume" : 2,
+        "cutting" : 0,
+        "warmth" : 50,
+        "phase" : "solid",
+        "environmental_protection" : 20,
```

Comment:
Environmental protection itself does nothing to protect from radiation. It protects from acid and gases.
For radiation, you need the radproof flag, like hazmat suit.